homeassistant/components/light/yeelight.py

```diff
@@ -6,157 +6,311 @@
+
+            self._rgb = self._get_rgb_from_properties()
+
+            self._available = True
+        except (yeelight.BulbException, socket.timeout):
+            self._available = False
+
+    @_cmd
+    def set_brightness(self, brightness, duration) -> None:
+        """Set bulb brightness."""
+        if brightness:
+            _LOGGER.debug("Setting brightness: %s", brightness)
+            self._bulb.set_brightness(brightness / 255 * 100,
+                                      duration=duration)
+
+    @_cmd
+    def set_rgb(self, rgb, duration) -> None:
+        """Set bulb's color."""
+        if rgb and self.supported_features & SUPPORT_RGB_COLOR:
+            _LOGGER.debug("Setting RGB: %s", rgb)
+            self._bulb.set_rgb(rgb[0], rgb[1], rgb[2], duration=duration)
+
+    @_cmd
+    def set_colortemp(self, colortemp, duration) -> None:
+        """Set bulb's color temperature."""
+        if colortemp and self.supported_features & SUPPORT_COLOR_TEMP:
+            temp_in_k = mired_to_kelvin(colortemp)
+            _LOGGER.debug("Setting color temp: %s K", temp_in_k)
+
+            self._bulb.set_color_temp(temp_in_k, duration=duration)
+
+    @_cmd
+    def set_default(self) -> None:
+        """Set current options as default."""
+        self._bulb.set_default()
+
+    @_cmd
+    def set_flash(self, flash) -> None:
+        """Activate flash."""
+        if flash:
+            from yeelight import RGBTransition, SleepTransition, Flow
+            if self._bulb.last_properties["color_mode"] != 1:
+                _LOGGER.error("Flash supported currently only in RGB mode.")
+                return
+
+            transition = self.config[CONF_TRANSITION]
+            if flash == FLASH_LONG:
+                count = 1
+                duration = transition * 5
+            if flash == FLASH_SHORT:
+                count = 1
+                duration = transition * 2
+
+            red, green, blue = self.rgb_color
+
+            transitions = list()
+            transitions.append(
+                RGBTransition(255, 0, 0, brightness=10, duration=duration))
+            transitions.append(SleepTransition(
+                duration=transition))
+            transitions.append(
+                RGBTransition(red, green, blue, brightness=self.brightness,
+                              duration=duration))
+
+            flow = Flow(count=count, transitions=transitions)
+            self._bulb.start_flow(flow)
+
+    def turn_on(self, **kwargs) -> None:
+        """Turn the bulb on."""
+        brightness = kwargs.get(ATTR_BRIGHTNESS)
+        colortemp = kwargs.get(ATTR_COLOR_TEMP)
+        rgb = kwargs.get(ATTR_RGB_COLOR)
+        flash = kwargs.get(ATTR_FLASH)
+
+        config_transition = self.config[CONF_TRANSITION]  # in ms
+        kwargs_transition = kwargs.get(ATTR_TRANSITION, 0) * 1000  # kwarg in s
+        duration = max(config_transition, kwargs_transition)
```

Comment:
One minor note here, kwargs_transition should take precedence even if it's smaller than config_transition. You can check if it's present by `if ATTR_TRANSITION in kwargs:`. Otherwise you've got the right idea here.